doc/RELEASE_NOTES.md

```diff
@@ -0,0 +1,828 @@
+
+Prior to v0.71, tool registration for remote hosts was recorded locally, and remotely via ssh.
+
+The registered tools are recorded in a local directory off of the "pbench_run" directory, by default `/var/lib/pbench-agent/tools-v1-<name>`, where `<name>` is the name of the Tool Group under which the tools were registered.
+
+The process of registering tools on local or remote hosts no longer validates that those tools are available during tool registration.  The Tool Meister sub-system now reports when registered tools are not present on registered hosts before beginning a benchmark run.  An error message will be displayed, and the particular "bench-script" will exit with a failure code.
+
+All tools registered prior to installing `v0.71.0-alpha` must be re-registered; tools registered locally or remotely on a host with v0.69 or earlier of the `pbench-agent` will be ignored.
+
+
+## New Support for Prometheus and PCP-based Tools
+
+The new "Tool Meister" sub-system enables support of Prometheus and PCP-based tools for data collection.
+
+The existing tools supported prior to the v0.71 release can be categorized as "Transient" tools.  By _transient_ we mean that a given tool is started and stopped immediately around the execution of a benchmark workload.  For example, when using `pbench-fio -b 4,16,32 -t read,write`, the transient tools are started immediately before each `fio` job is executed, and stopped immediately following its completion, for each of the 6 (six) `fio` jobs that would be run.
+
+A new category is introduced for Prometheus and PCP called "Persistent" tools. Persistent are started once at the beginning of a benchmark script, stopped at its end.  Using the previous `pbench-fio` example, persistent tools would be started before any of the 6 (six) `pbench-fio` jobs begin, and would be stopped once all six end.
+
+When persistent tools are used, data is continuously collected from the data sources ("exporters", in the case of Prometheus, and "PMCDs", in the case of PCP) and stored local to the execution of the Tool Data Sink.
+
+Note that for transient tools, where data for the transient tool is collected locally on the host the tool is registered, the collected data is sent to the Tool Data Sink when the benchmark script deems it won't impact behavior of the benchmark itself.
+
+### Prometheus tools: `node-exporter` and `dcgm`
+
+Two new pbench "tools" have been added, `node-exporter` and `dcgm`.  If one registers either or both of these new tools (e.g. via `pbench-register-tools --name=node-exporter`), then the Tool Meister sub-system will run the `node_exporter` code on the registered hosts, and a local instance of Prometheus to collect the data.  The collected Prometheus data is stored in the pbench result directory as a tar ball at: `${pbench_run}/<script>_<config>_YYYY.MM.DDTHH.mm.ss/tools-<group>/prometheus`.
```

Comment:
> If one registers either

I would suggest the passive voice rather than the indefinite active; e.g.,

> If either or both of these new tools is registered (e.g., via [...]), then ...

> hosts, and a local instance of Prometheus to collect the data

The comma should be removed.  Also, I would add a word or two to underscore the fact that the data is stored locally to the TM during the run and transferred to the TDS later.  (We're a little too quick to move on to outlining how the data is added to the result directory.)

Reply:
Implied is the behavior of Prometheus which "scrapes" data at the requested interfaces and stores that data in its local prometheus database.